src/Source/FileSource.js

```diff
@@ -145,10 +146,32 @@ class FileSource extends Source {
         return this.url;
     }
 
-    onParsedFile(parsedFile) {
-        this.parsedData = parsedFile;
-        this.extent = parsedFile.extent;
-        return parsedFile;
+    onLayerAdded(options) {
+        super.onLayerAdded(options);
+        let parsedData = this._parsedDatasCaches[options.crsOut].getByArray([0]);
+        if (!parsedData) {
+            options.buildExtent = true;
+            parsedData = this.parser(this.fetchedData, options);
```

Comment:
Is `this.fetchedData` always present at this time ?

Reply:
yes, `onLayerAdded` is called after `source.whenReady` resolving